tests/mocks.py

```diff
@@ -96,6 +97,50 @@ def validate_change(change):
     return True
 
 
+def get_stack(_phid, phabdouble):
+    phids = set()
+    new_phids = {_phid}
+    edges = []
+
+    # Repeatedly request all related edges, adding connected revisions
+    # each time until no new revisions are found.
+    # NOTE: this was adapted from previous implementation of build_stack_graph.
+    while new_phids:
+        phids.update(new_phids)
+        edges = [
+            e
+            for e in phabdouble._edges
+            if e["sourcePHID"] in phids
+            and e["edgeType"] in ("revision.parent", "revision.child")
+        ]
+        new_phids = set()
+        for edge in edges:
+            new_phids.add(edge["sourcePHID"])
+            new_phids.add(edge["destinationPHID"])
+
+        new_phids = new_phids - phids
+
+    # Treat the stack like a commit DAG, we only care about edges going
+    # from child to parent. This is enough to represent the graph.
+    edges = {
+        (edge["sourcePHID"], edge["destinationPHID"])
+        for edge in edges
+        if edge["edgeType"] == "revision.parent"
+    }
+
+    stack_graph = defaultdict(list)
+    for edge in edges:
```

Comment:
Here we're always using `edge` with some index - can we just unpack it in the loop so it's more obvious what `edge[0]` and `edge[1]` are supposed to represent?

Reply:
Done, + added a `sources` list to make checking for root node a little more readable.